spec/lib/sc_client_spec.rb

```diff
@@ -0,0 +1,131 @@
+require 'spec_helper'
+
+RSpec.describe Socketclusterclient do
+  include Socketclusterclient
+
+  describe 'socketcluster client' do
+    describe 'initialize' do
+      before(:each) do
+        @connect_url = 'ws://localhost:8000/socketcluster/'
+        @socket = ScClient.new(@connect_url)
+      end
+
+      context 'instance variables' do
+        it 'should initialize all instance variables' do
+          expect(@socket.instance_variable_get(:@id)).to eq('')
+          expect(@socket.instance_variable_get(:@cnt)).to eq(0)
+          expect(@socket.instance_variable_get(:@acks)).to eq({})
+          expect(@socket.instance_variable_get(:@channels)).to eq([])
+          expect(@socket.instance_variable_get(:@enable_reconnection)).to eq(true)
+          expect(@socket.instance_variable_get(:@delay)).to eq(3)
+        end
+
+        it 'should assign connect_url to instance variable @url' do
+          expect(@socket.instance_variable_get(:@url)).to eq(@connect_url)
+        end
+      end
+    end
+
+    describe 'set_basic_listener' do
+      let(:on_connect) { -> { puts 'on connect got called' } }
+      let(:on_disconnect) { -> { puts 'on connect got called' } }
+      let(:on_connect_error) { -> { puts 'on connect got called' } }
+
+      before(:each) do
+        @connect_url = 'ws://localhost:8000/socketcluster/'
```

Comment:
You can remove this line and put it in before(:each) block in scope of 'socketcluster client' describe block

Reply:
Done.